controllers/ArticleController.js

```diff
@@ -80,32 +80,76 @@ class ArticleController {
   }
 
   /**
-   * get all articles created
+   * get all articles created and use the query
+   * if provided to implement pagination
    * @param {object} req - request object
    * @param {object} res - response object
    * @returns {object} - the found article from database or empty if not found
    */
   static listAllArticles(req, res) {
+    /** get the page number from the query
+     *  and if not provided make it a default
+     *  value  of 1
+     */
+    let { page } = req.query;
+    let limit, offset;
+
+    if (page) {
+      /** convert the page query to an integer
+       *  check that the integer is greater
+       *  than zero and is not a float or string (NaN)
+       */
+      page = Number(page);
+      /** calculate offset and limit */
+      limit = 10;
+      offset = limit * (page - 1);
+
+      if (page < 1 || !Number.isInteger(page)) {
```

Comment:
Please move this from controllers, it should be handled by a middleware function.

Reply:
okay